dev/archery/archery/linking.py

```diff
@@ -61,9 +63,84 @@ def list_dependency_names(self):
             names.append(name)
         return names
 
+    def _extract_symbols(self, symbol_info):
+        return [re.search(r'\S+$', line).group() for line in symbol_info if line]
+
+    def _remove_weak_symbols(self, symbol_info):
+        return [line for line in symbol_info if not re.search(r'\s[Ww]\s', line)]
+
+    def _capture_symbols(self, remove_symbol_versions, symbol_info):
+        if remove_symbol_versions:
+            symbol_info = [re.split('@@', line)[0] for line in symbol_info]
```

Comment:
1. You don't need a regular expression for this, do you?
2. Why the double '@@'? Shouldn't it be '@'?

Reply:
Yes you're probably correct here, I think there were some symbols with @@, 
```bash
00000000000147a0 T _dl_allocate_tls@@GLIBC_PRIVATE
```
But again, we can just get the left with a single `@`. You're right.